Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -644,6 +625,67 @@ public final class SwiftTargetBuildDescription {
 
         return args
     }
+    
+    /// Determines the arguments needed to run `swift-symbolgraph-extract` for
+    /// this module.
+    public func symbolGraphExtractArguments() throws -> [String] {
+        var args = [String]()
+        args += try self.cxxInteroperabilityModeArguments(allowDuplicate: true)
+        return args
+    }
+
+    /// Determines the arguments needed for cxx interop for this module.
+    ///
+    /// If the current module or any of its linked dependencies requires cxx
+    /// interop, cxx interop will be enabled on the current module.
```

Comment:
We don't actually want to implicitly enable C++ interop for anything other than the test discovery target or the symbolgraph target.
One of the reasons for that is that C++ interop is incompatible with Swift's library evolution. If a dependency of your project uses C++ internally, without re-exporting any C++ symbols, you don't need to enable C++ interop and can keep library evolution enabled.